src/Campaigns/Actions/CacheCampaignData.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\CampaignsDataQuery;
+use Give\Campaigns\Models\Campaign;
+use Give\Donations\Models\Donation;
+
+/**
+ * @unreleased
+ *
+ * @uses givewp_donation_created hook
+ * @uses givewp_donation_updated hook
+ *
+ * Action used to update campaign's stats data
+ *
+ */
+class CacheCampaignData
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(Donation $donation): void
+    {
+        if ( ! $donation->status->isComplete()) {
+            return;
+        }
+
+        as_enqueue_async_action('givewp_cache_campaign_data', [$donation->formId], 'givewp_campaigns_cache');
```

Comment:
Is there any reason we can't just pass the `$donation->campaignId` directly?  It should be [available](https://github.com/impress-org/givewp/blob/ac3ca0772449492f40cee289d4eb8c11e21142f1/src/Donations/Repositories/DonationRepository.php#L433) by the time the model action is fired.

Alternatively we could just pass the `$donation->id` and use the donation model in the handle cache logic as that would be more flexible for the future